gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -964,21 +986,32 @@ public void onFailure(GoogleJsonError jsonError, HttpHeaders responseHeaders) {
   // Please check that removing it is correct, and remove this comment along with it.
   // @VisibleForTesting
   public static void validateCopyArguments(
-      String srcBucketName,
-      List<String> srcObjectNames,
-      String dstBucketName,
-      List<String> dstObjectNames,
+      List<StorageResourceId> srcObjects,
+      List<StorageResourceId> dstObjects,
       GoogleCloudStorage gcsImpl)
       throws IOException {
+
+    checkNotNull(srcObjects, "srcObjects must not be null");
+    checkNotNull(dstObjects, "dstObjects must not be null");
     Preconditions.checkArgument(
-        !isNullOrEmpty(srcBucketName), "srcBucketName must not be null or empty");
-    Preconditions.checkArgument(
-        !isNullOrEmpty(dstBucketName), "dstBucketName must not be null or empty");
-    Preconditions.checkArgument(srcObjectNames != null, "srcObjectNames must not be null");
-    Preconditions.checkArgument(dstObjectNames != null, "dstObjectNames must not be null");
-    Preconditions.checkArgument(
-        srcObjectNames.size() == dstObjectNames.size(),
-        "Must supply same number of elements in srcObjectNames and dstObjectNames");
+        srcObjects.size() == dstObjects.size(),
+        "Must supply same number of elements in srcObjects and dstObjects");
+
+    if (srcObjects.isEmpty()) {
+      return;
+    }
+
+    String srcBucketName = srcObjects.get(0).getBucketName();
+    String dstBucketName = dstObjects.get(0).getBucketName();
+
+    for (int i = 0; i < srcObjects.size(); i++) {
```

Comment:
While this check makes sense in default implementation in the interface, it maybe too strict here?